packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/secretValuesWalkthrough.ts

```diff
@@ -176,12 +176,20 @@ const enterSecretName = async (invalidNames: string[]) =>
 
 const secretValueDefaultMessage = (secretName: string) => `Enter the value for ${secretName}:`;
 
+export const secretValueValidator = () => (input?: string) => {
+  if (!input || input.length === 0 || input.length > 2048) {
```

Comment:
Are the `!input` and `input.length === 0` checks both necessary?

Reply:
I think `input.length === 0` should be enough, but I added `!input` just in case `input` got a value of `undefined`

Reply:
Instead of `!input`, can you use `typeof input !== 'string'`.

Reply:
Sure, no problem